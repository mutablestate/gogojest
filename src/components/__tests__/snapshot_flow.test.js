import React from 'react'
import renderer from 'react-test-renderer'

import SnapshotFlow from '../snapshot_flow'

test('JSX matches snapshot', () => {
  const component = renderer.create(<SnapshotFlow />)
  expect(component).toMatchSnapshot()
})
