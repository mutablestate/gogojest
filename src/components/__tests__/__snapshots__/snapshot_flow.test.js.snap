exports[`test JSX matches snapshot 1`] = `
<div>
  <h1>
    Snapshot Workflow
  </h1>
  <h4>
    To creat a component test
  </h4>
  <ol>
    <li>
      Add a test file 
      <span
        style={
          Object {
            "color": "#ca461a",
            "fontFamily": "monospace",
            "fontStyle": "italic",
            "fontWeight": "bold",
          }
        }>
        [component_name].test.js
      </span>
       to a 
      <span
        style={
          Object {
            "color": "#ca461a",
            "fontFamily": "monospace",
            "fontStyle": "italic",
            "fontWeight": "bold",
          }
        }>
        __tests__
      </span>
       directory close to the component code.
    </li>
    <li>
      Use 
      <span
        style={
          Object {
            "color": "#ca461a",
            "fontFamily": "monospace",
            "fontStyle": "italic",
            "fontWeight": "bold",
          }
        }>
        toMatchSnapshot()
      </span>
       to assert the rendered component matches the snapshot
    </li>
    <li>
      Running the test creates a snapshot file in the 
      <span
        style={
          Object {
            "color": "#ca461a",
            "fontFamily": "monospace",
            "fontStyle": "italic",
            "fontWeight": "bold",
          }
        }>
        src/__tests__/components/__snapshots__
      </span>
       directory
    </li>
    <li>
      When the test passes 
      <span
        style={
          Object {
            "color": "#ca461a",
            "fontFamily": "monospace",
            "fontStyle": "italic",
            "fontWeight": "bold",
          }
        }>
        git commit
      </span>
       the snapshot and component
    </li>
  </ol>
  <h4>
    Updating anything in a component breaks the snapshot
  </h4>
  <ol>
    <li>
      View the snapshot diff in the watcher
    </li>
    <li>
      To accept the changes: update the snapshot file (type
      <span
        style={
          Object {
            "color": "#ca461a",
            "fontFamily": "monospace",
            "fontStyle": "italic",
            "fontWeight": "bold",
          }
        }>
        u
      </span>
       in the watcher) and commit the updated snapshot and component
    </li>
    <li>
      To discard the changes: undo the component update to match the snapshot
    </li>
  </ol>
</div>
`;
